test/test-generate-request-details.js

```diff
@@ -243,4 +244,26 @@ suite('Test Generate Request Details', function() {
     assert.equal(details.headers.Topic, extraOptions.headers.Topic);
     assert.equal(details.headers.Urgency, extraOptions.headers.Urgency);
   });
+
+  test('Audience contains port', function() {
+    const subscription = {
+      endpoint: 'http://example.com:4242/life-universe-and-everything'
+    };
+
+    const extraOptions = {
+      vapidDetails: {
+        subject: 'mailto:example@example.com',
+        publicKey: vapidKeys.publicKey,
+        privateKey: vapidKeys.privateKey
```

Comment:
Right, I forgot it was already doing that.

It's unexpected then that it doesn't work. The failing test was saying that the key was not of the correct length, but it shouldn't be happening (we even have a test for this: https://github.com/web-push-libs/web-push/blob/master/test/test-vapid-helper.js#L29).

@gauntface any idea?